erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -993,7 +995,7 @@ def get_transfered_raw_materials(self):
 				filters={'parent': self.work_order, 'item_code': item_code},
 				fields=["required_qty", "consumed_qty"]
 				)
-			if not req_items:
+			if backflush_based_on=="BOM" and not req_items:
```

Comment:
This will not be "BOM", because this function is only called when the backflush setting is set to "Material Transferred for Manufacture".